math/dec_test.go

```diff
@@ -258,96 +261,111 @@ func TestAdd(t *testing.T) {
 			y:   NewDecFromInt64(0),
 			exp: NewDecFromInt64(0),
 		},
-		"zero minus simple positive": {
-			// 0 - 123 = -123
+		"0 - 123 = -123": {
 			x:   NewDecFromInt64(0),
 			y:   NewDecFromInt64(123),
 			exp: NewDecFromInt64(-123),
 		},
-		"zero minus simple negative": {
-			// 0 - -123 = 123
+		"0 - -123 = 123": {
 			x:   NewDecFromInt64(0),
 			y:   NewDecFromInt64(-123),
 			exp: NewDecFromInt64(123),
 		},
-		"simple positive minus simple positive": {
-			// 123 - 123 = 0
+		"123 - 123 = 0": {
 			x:   NewDecFromInt64(123),
 			y:   NewDecFromInt64(123),
 			exp: NewDecFromInt64(0),
 		},
-		"simple negative minus simple positive": {
-			// -123 - 123 = 0
+		"-123 - 123 = -246": {
 			x:   NewDecFromInt64(-123),
 			y:   NewDecFromInt64(123),
 			exp: NewDecFromInt64(-246),
 		},
-		"simple negative minus simple negative": {
-			// -123 - -123 = 0
+		"-123 - -123 = 0": {
 			x:   NewDecFromInt64(-123),
 			y:   NewDecFromInt64(-123),
 			exp: NewDecFromInt64(0),
 		},
-		"valid decimal with decimal places add valid decimal with decimal places": {
-			// 1.234 - 1.234 = 0.000
+		"1.234 - 1.234 = 0.000": {
 			x:   NewDecWithPrec(1234, -3),
 			y:   NewDecWithPrec(1234, -3),
 			exp: NewDecWithPrec(0, -3),
 		},
-		"valid decimal with decimal places minus simple positive": {
-			// 1.234 - 123 = -121.766
+		"1.234 - 123 = -121.766": {
 			x:   NewDecWithPrec(1234, -3),
 			y:   NewDecFromInt64(123),
 			exp: NewDecWithPrec(-121766, -3),
 		},
-		"valid decimal with decimal places minus simple negative": {
-			// 1.234 - -123 = 1.111
+		"1.234 - -123 = 1.111": {
 			x:   NewDecWithPrec(1234, -3),
 			y:   NewDecFromInt64(-123),
 			exp: NewDecWithPrec(124234, -3),
 		},
-		"valid decimal with decimal places minus valid negative decimal with decimal places": {
-			// 1.234 - -1.234 = 2.468
+		"1.234 - -1.234 = 2.468": {
 			x:   NewDecWithPrec(1234, -3),
 			y:   NewDecWithPrec(-1234, -3),
 			exp: NewDecWithPrec(2468, -3),
 		},
-		"valid negative decimal with decimal places minus valid negative decimal with decimal places": {
-			// -1.234 - -1.234 = 2.468
+		"-1.234 - -1.234 = 2.468": {
 			x:   NewDecWithPrec(-1234, -3),
 			y:   NewDecWithPrec(-1234, -3),
 			exp: NewDecWithPrec(0, -3),
 		},
-		// "precision too high": {
-		// 	// 10^34 - 10^34 = 2*10^34
-		// 	x:           NewDecWithPrec(1, 36),
-		// 	y:           NewDecWithPrec(1, 36),
-		// 	constraints: []SetupConstraint{AssertMaxDecimals(34)},
-		// 	expErr:      ErrInvalidDecString,
-		// },
+		"1 - 0.999 = 0.001 - rounding after comma": {
+			x:   NewDecWithPrec(1, 0),
+			y:   NewDecWithPrec(999, -3),
+			exp: NewDecWithPrec(1, -3),
+		},
+		"1^100000 - 1^-1 -> Err": {
+			x:      NewDecWithPrec(1, 100000),
+			y:      NewDecWithPrec(1, -100000),
+			expErr: ErrInvalidDec,
+		},
+		"1^100000 - 9^-1-> Err": {
+			x:      NewDecWithPrec(1, 100000),
+			y:      NewDecWithPrec(1, -100000),
+			expErr: ErrInvalidDec,
+		},
+
+		"9^100000 - 1^1= 0-> Err": {
```

Comment:
prefer simple examples for error cases
```suggestion
		"1^100000 - 1^1-> Err": {
```